### PR TITLE
per-file: Put folder before files and add basic searching

### PR DESCRIPTION
**Pull request by davidot**
Please review my colours as I'm probably the worst designer in the world.
I only added the warning if you have too many results.
Dark mode:
libjs-website-warn1

Light mode:
libjs-website-warn2

### CODE REVIEW

File: test262/per-file/main.js [hunk 1]
```diff
@@ -179,37 +181,136 @@ function generateChildren(node) {
...
+        ];
+        makeChildNavigable(childNode, extraPathParts, targetNode);
       }
+
+      targetNode.appendChild(childNode);
     });
+
+  if (foundResults.length > maxResultsShown) {
+    const extraNode = document.createElement("p");
+    extraNode.innerText = `Only displaying the first ${maxResultsShown} results! In total had ${foundResults.length} results.`;
```
Comment (davidot) [inline test262/per-file/main.js:310]: Hmm maybe it should be `In total _have found_ 999 results` or something like that?

File: test262/per-file/main.js [hunk 2]
```diff
@@ -153,6 +153,12 @@ function generateQueryString(pathSegments) {
   return `?path=${pathSegments.join("/")}`;
 }
 
+function sortResultsByTypeAndName([nameLhs, lhsResult], [nameRhs, rhsResult]) {
```
Comment (linusg) [inline test262/per-file/main.js]: Shouldn't this be `lhsName`, `rhsName`?

File: test262/per-file/main.js (hunk 1 above)
Comment (linusg) [reply test262/per-file/main.js]: "Only showing the first X of Y total results"?

**Review by linusg — COMMENTED** [review]
Sorry, completely forgot about this PR! Please ping more aggressively :^)

File: test262/per-file/main.js (hunk 1 above)
Comment (davidot) [reply test262/per-file/main.js]: Yeah that's better

File: test262/per-file/main.js (hunk 2 above)
Comment (davidot) [reply test262/per-file/main.js]: Fixed.

**Comment by davidot** [comment]
I added a checkbox for toggling filter vs. searching. (Also maybe the result type filtering is a bit hidden but :shrug:)

**Pull request merged**